src/main/java/org/elasticsearch/index/mapper/internal/SourceFieldMapper.java

```diff
@@ -420,19 +429,23 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     @Override
     public void merge(Mapper mergeWith, MergeResult mergeResult) throws MergeMappingException {
         SourceFieldMapper sourceMergeWith = (SourceFieldMapper) mergeWith;
-        if (!mergeResult.simulate()) {
+        if (mergeResult.simulate()) {
+            if (this.enabled != sourceMergeWith.enabled) {
+                mergeResult.addConflict("Cannot update enabled setting for [_source]");
+            }
+            if (Arrays.equals(this.includes, sourceMergeWith.includes) == false) {
```

Comment:
Good point, I'll change to `.includes()` and `.excludes()` which handle null, and add tests.